notebooks/working/Simple_Simulation.md

```diff
@@ -0,0 +1,382 @@
+
+```python
+data = pd.read_csv('spring_2016_all_bay_area_long_format_plus_cross_bay_col.csv')
+data.columns
+```
+
+```python
+### Just look at the drive alone mode
+drive_alone_df = data[data['mode_id']==1]
+
+### Only keep columns of interest 
+drive_alone_df = drive_alone_df[[ 'total_travel_time', 'total_travel_cost',
+                                 'total_travel_distance', 'household_size',
+                               'num_cars', 'cross_bay']]
+
+drive_alone_df.describe()
+
+```
+
+## Assumed causal graph
+
+```python
+drive_alone_graph = CausalGraphicalModel(
+    nodes=['total_travel_time', 'total_travel_cost', 'total_travel_distance', #'household_income',
+          'household_size',  'num_cars', 'cross_bay', 'utility_driving'],
+    edges=[
+        ("total_travel_time", "utility_driving"), 
+        ("total_travel_cost", "utility_driving"), 
+        ("total_travel_distance", "utility_driving"), 
+        ("household_size", "utility_driving"), 
+#         ("household_income", "utility_driving"), 
+        ("num_cars", "utility_driving"), 
+        ("cross_bay", "utility_driving"), 
+        
+        
+        ("total_travel_distance", "total_travel_time"), 
+        ("total_travel_distance", "total_travel_cost"), 
+    ]
+)
+
+# draw return a graphviz `dot` object, which jupyter can render
+drive_alone_graph.draw()
+```
+
+## Distributional regression
+Assume univariate linear approximation for the relationship between travel distance and travel time/cost. Turns out it's not a horrible assumption.
+
+```python
+def fit_regression(X, y, data, plotting = True):
+    data_x = sm.add_constant(data[X])
+    data_y = data[y]
+
+    model = sm.OLS(data_y, data_x)
+    
+    results = model.fit()
+    
+    if plotting:
+        fig = plt.figure(figsize=(12,8))
+        fig = sm.graphics.plot_regress_exog(results, X[0], fig=fig)
+    
+    return results
+```
+
+```python
+## Regress travel cost on travel distance
+
+cost_on_distance_reg = fit_regression(X=['total_travel_distance'], 
+                                      y = ['total_travel_cost'], data=drive_alone_df, plotting = True)
+
+cost_on_distance_reg.summary()
+```
+
+```python
+## Regress travel time on travel distance
+
+time_on_distance_reg = fit_regression(X=['total_travel_distance'], 
+                                      y = ['total_travel_time'], data=drive_alone_df, plotting = True)
+
+time_on_distance_reg.summary()
+```
+
+### Simulate travel time and cost based on the distributional regression
+
+```python
+## residuals spread -- assuming homoscedasticity
+time_on_dist_std = time_on_distance_reg.resid.std()
+cost_on_dist_std = cost_on_distance_reg.resid.std()
+
+sample_size = len(drive_alone_df)
+```
+
+```python
+total_travel_time_sim = ( time_on_distance_reg.params[0] + 
+                         time_on_distance_reg.params[1]*drive_alone_df['total_travel_distance']
+                        + np.random.normal(loc=0, scale=time_on_dist_std, size = sample_size) )
```

Comment:
Before simulating the various variables, I would suggest adding a call to `np.random.seed` to make the results of the simulations reproducible.